docs/writing-tests/testharness-api.md

```diff
@@ -248,6 +248,21 @@ promise_test(function(t) {
 }, "Another example");
 ```
 
+## Asynchronous Promise Tests ##
+
+`promise_test` are running sequentially. This might causes timeout when too many
+have to run. `async_promise_test` is a variant, running in parallel.
+
+|              | in sequence | in parallel       |
+| -------------| ----------: | ----------------: |
+| **Function** | test        | async_test        |
+| **Promise**  | promise_test| async_promise_test|
+
+It uses the same API as `promise_test`:
+```js
+async_promise_test(test_function, name, properties);
+```
+
```

Comment:
Ideally, we'd document the relationship to `setup()` and `promise_setup()` as well. Could you take a stab at that?

Reply:
Done